File: Examples/VisualBasic/Images/ExtractImages.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic
Imports System.Drawing
Namespace Images
    Public Class ExtractImages
        Public Shared Sub Run()
            Try
                ' ExStart:ExtractImages
                ' The path to the documents directory.
                Dim dataDir As String = RunExamples.GetDataDir_Images()

                ' Load the document into Aspose.Note.
                Dim oneFile As New Document(dataDir & Convert.ToString("Aspose.one"))

                ' Get all Image nodes
                Dim nodes As IList(Of Aspose.Note.Image) = oneFile.GetChildNodes(Of Aspose.Note.Image)()

                For Each image As Aspose.Note.Image In nodes
                    Using stream As New MemoryStream(image.Bytes)
                        Using bitMap As New Bitmap(stream)
                            ' Save image bytes to a file
                            bitMap.Save([String].Format(dataDir & Convert.ToString("{0}"), Path.GetFileName(image.FileName)))
                        End Using
                    End Using
                    ' ExEnd:ExtractImages 
                Next
            Catch ex As Exception
                Console.WriteLine(ex.Message)
                Console.ReadLine()
            End Try
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Images/ExtractImages.cs
using System.IO;
using Aspose.Note;
using System.Collections.Generic;
using System.Drawing;
using System;

namespace Aspose.Note.Examples.CSharp.Images
{
    public class ExtractImages
    {
        public static void Run()
        {
            try
            {
                // ExStart:ExtractImages
                // ExFor:Document
                // ExFor:CompositeNode`1.GetChildNodes
                // ExFor:Image
                // ExFor:Image.Bytes
                // ExSummary:Shows how to get an image from a document.

                // The path to the documents directory.
                string dataDir = RunExamples.GetDataDir_Images();

                // Load the document into Aspose.Note.
                Document oneFile = new Document(dataDir + "Aspose.one");

                // Get all Image nodes
                IList<Aspose.Note.Image> nodes = oneFile.GetChildNodes<Aspose.Note.Image>();

                foreach (Aspose.Note.Image image in nodes)
                {
                    using (MemoryStream stream = new MemoryStream(image.Bytes))
                    {
                        using (Bitmap bitMap = new Bitmap(stream))
                        {
                            // Save image bytes to a file
                            bitMap.Save(String.Format(dataDir + "{0}", Path.GetFileName(image.FileName)));
                        }
                    }
                }

                // ExEnd:ExtractImages 
            }
            catch (Exception ex)
            {
                Console.WriteLine("\r\nError: " + ex.Message + "\r\nPress any key to continue.");
                Console.ReadLine();
            }
        }
    }
}